jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
+
+        List<String> values=_preferredEncodingOrderCache.get(key);
+        if (values==null)
+        {
+            QuotedEncodingQualityCSV encodingQualityCSV = new QuotedEncodingQualityCSV(_precompressedFormats);
+            Enumeration<String> e2=request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
```

Comment:
We should be able to avoid this second enumeration of the accept encodings.   If we really have to enumerate over them to build the key, they you may as well use "," as the separator in the key and then just use the key that missed ti build the QuotedEncodingQualityCSV instance.